spring-cloud-task-batch/src/test/java/org/springframework/cloud/task/batch/listener/TaskBatchExecutionListenerTests.java

```diff
@@ -91,6 +97,17 @@ public void testAutobuiltDataSource() {
 	}
 
 	@Test
+	public void testBeanFactoryReturnsSameInstance() throws Exception{
```

Comment:
creates a `TaskBatchExecutionListener` instance and it is returned.  The second getObject call return the previously created instance.